### PR TITLE
util: add number codec

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @coocood @qiuyesuifeng PTAL

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

File: src/util/codec/number.rs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+/// `encode_u64_desc` guarantees that the encoded value is in descending order for comparison.
+pub fn encode_u64_desc(buf: &mut [u8], v: u64) -> Result<()> {
+    try!(check_bound(buf, 8));
+    BigEndian::write_u64(buf, !v);
+    Ok(())
+}
+
+// `decode_u64` decodes value encoded by `encode_u64` before.
+// It returns the leftover un-decoded slice, decoded value if no error.
+pub fn decode_u64(buf: &[u8]) -> Result<u64> {
```
Comment (qiuyesuifeng) [inline src/util/codec/number.rs]: Seems no return value for  un-decoded slice, out of date comment.

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**